thunder/executors/torchex.py

```diff
@@ -421,12 +421,113 @@ def _tensor_from_sequence_prims_transform(
     return tensor_from_sequence(seq_or_number, device=torch_device, dtype=torch_dtype)
 
 
+def _set_rng_state_prim_impl(new_state: torch.Tensor, device: torch.device | None) -> torch.Tensor:
+    if device is None:
+        device = torch.cuda.current_device()
+    torch.cuda.set_rng_state(new_state, device)
+    return new_state
+
+
+set_rng_state_prim_impl = ex.register_operator(
+    "set_rng_state_prim_impl",
+    meta=prims.set_rng_state.meta,
+    fn=_set_rng_state_prim_impl,
+    tags=(prims.OpTags.RANDOM_OP, prims.OpTags.DONT_DCE),
+)
+
+
+def _get_rng_state_prim_impl(state: torch.Tensor | None, device: torch.device | None) -> torch.Tensor:
+    if state is not None:
+        return state
+    if device is None:
+        device = torch.cuda.current_device()
+    state = torch.cuda.get_rng_state(device)
+    return state
+
+
+get_rng_state_prim_impl = ex.register_operator(
+    "get_rng_state_prim_impl",
+    meta=prims.get_rng_state.meta,
+    fn=_get_rng_state_prim_impl,
+    tags=(prims.OpTags.RANDOM_OP,),
+)
+
+
+def _unpack_rng_state_prim_torch_impl(s: torch.Tensor) -> tuple[int, int]:
+    seed, offset = torch.chunk(s, 2)
+    return seed.view(torch.int64).item(), offset.view(torch.int64).item()
+
+
+unpack_rng_state_prim_torch_impl = ex.register_operator(
+    "unpack_rng_state_prim_torch_impl",
+    meta=prims.unpack_rng_state.meta,
+    fn=_unpack_rng_state_prim_torch_impl,
+    tags=(prims.OpTags.RANDOM_OP,),
+)
+
+
+def _unpack_rng_state_prim_nvfuser_impl(s: torch.Tensor) -> tuple[int, int]:
+    seed, offset = torch.chunk(s, 2)
+    return seed.view(torch.int64).item(), offset.view(torch.int64).item() // 4
+
+
+unpack_rng_state_prim_nvfuser_impl = ex.register_operator(
+    "unpack_rng_state_prim_nvfuser_impl",
+    meta=prims.unpack_rng_state.meta,
+    fn=_unpack_rng_state_prim_nvfuser_impl,
+    tags=(prims.OpTags.RANDOM_OP,),
+)
+
+
+def _update_rng_state_prim_torch_impl(seed: int, offset: int) -> torch.Tensor:
+    # NOTE: Pytorch updates the offset based on the shape of the input,
+    # https://github.com/pytorch/pytorch/blob/a8aed4ce3f67aa4236e80a11209a40dd0c4ebc87/aten/src/ATen/native/cuda/DistributionTemplates.h#L39-L63,
+    # rather than increasing it by exactly 4 each time.
+    new_offset = offset + 4
+    seed = torch.tensor(seed)
+    offset = torch.tensor(new_offset)
+    seed_portion = seed.reshape([1]).view(torch.uint8)
+    offset_portion = offset.reshape([1]).view(torch.uint8)
+    new_state = torch.cat([seed_portion, offset_portion])
+    return new_state
+
+
+update_rng_state_prim_impl = ex.register_operator(
+    "pack_rng_state_prim_impl",
+    meta=prims.update_rng_state.meta,
+    fn=_update_rng_state_prim_torch_impl,
+    tags=(prims.OpTags.RANDOM_OP,),
+)
+
+
+def _update_rng_state_prim_nvfuser_impl(seed: int, offset: int) -> torch.Tensor:
+    new_offset = (offset + 1) * 4
```

Comment:
This line requires an explanation of why offset is rounded to the nearest multiple of 4.

Reply:
we use the nvfuser way here and the transformation is only applied to nvfuserex as in the other [PR](https://github.com/Lightning-AI/lightning-thunder/blob/6b5a4aca7867476db6505dacdfe7bb6c3afa6929/thunder/executors/nvfuserex_impl.py#L750-L753)

Reply:
I meant that we should include the explanation as a comment in the code. I'm sorry for the confusion!